### DISCUSSION TITLE
How to create a new remote module

### DISCUSSION THREAD

**Discussion by monaqa** [nvim-treesitter/nvim-treesitter#6484] [Q&A] [answered]
I would like to create a plugin using nvim-treesitter, what is the current recommended process to follow?

The current master [CONTRIBUTING.md](https://github.com/nvim-treesitter/nvim-treesitter/blob/ea4f6d5175812a3f48302d638a766bc18fcbb04a/CONTRIBUTING.md) says:

> In any case, you can build your own module! To help you get started in the process, we have a template repository designed to build new modules [here](https://github.com/nvim-treesitter/module-template).

However, the linked module-template repository has been archived and says

> Nvim-treesitter modules are deprecated and will be removed; new tree-sitter based plugins should be standalone.

I am not sure what the "standalone" refers to.

**Comment by clason** [comment, marked as answer]
> I would like to create a plugin using nvim-treesitter, what is the current recommended process to follow?

Don't.

> I am not sure what the "standalone" refers to.

It means it should not rely on nvim-treesitter in any way. Just assume that users have _some_ way of installing parsers, and use Neovim's own `vim.treesitter` API to implement it. (See https://github.com/nvim-treesitter/nvim-treesitter-context for an example.)

**Reply by monaqa** [reply]
Thanks for the quick reply!
I will take a cue from https://github.com/nvim-treesitter/nvim-treesitter-context.
